### ISSUE TITLE
explore(NN)

### ISSUE THREAD

**Issue by zerline** [sagemath/sage-explorer#38]
explore(NN) crashes:

```
/home/sage/sage-explorer/sage_explorer/sage_explorer.py in reset(self)
    704         self.explorables = []
    705         for p in get_properties(self.value, Settings.properties):
--> 706             explorable = getattr(self.value, p.name).__call__()
    707             children.append(Box((Label(p.prop_label),), layout=Layout(border='1px solid #eee')))
    708             e = ExplorableCell(explorable, initial_value=self.value)

AttributeError: 'NoneType' object has no attribute 'category'
```

**Comment by zerline** [comment]
This bug comes from the fact that `NN` does not compare:

```
sage: if NN:
....:     print("ok")
....:  
```

If this Sage behavior is wished, then we'll have to compare to None everywhere needed. @nthiery What do you think? 

Please review e51147ad66f7e72e

**Comment by nthiery** [comment]
Thanks Odile for the analysis. That NN evaluates to false in boolean context is presumably a misfeature: NN being a collection, it seems natural to follow the convention of collections: evaluate to false iff the collection is empty. So we should create a ticket on Sage.

That being said, testing with "is None" is better: there is no reason to explore trivial values such as empty collections or zero elements in a different way from others. So +1 on the change.

**Comment by zerline** [comment]
Merged e51147ad66f7e72ea4c203ff78e3be1a411486b1
